src/main/java/org/apache/commons/math4/util/CompositeFormat.java

```diff
@@ -174,7 +177,7 @@ public static boolean parseFixedstring(final String source,
         final int endIndex = startIndex + expected.length();
         if ((startIndex >= source.length()) ||
             (endIndex > source.length()) ||
-            (source.substring(startIndex, endIndex).compareTo(expected) != 0)) {
+            (source.substring(startIndex, endIndex).compareTo(expected) != 0)) { // #1
```

Comment:
I would expect this call to `substring` to not issue an error, because the expected type for both arguments is `IndexOrHigh` - which should both be established by the conditionals, as your suppress warning comment notes. Can you write a test case like this code and submit a bug? Thanks!

Reply:
The explanation `@SuppressWarnings` was wrong, it actually shows `array.access.unsafe.low` error as it cannot recognise `pos.getIndex()` as `@NonNegative`.
The PR with ParsePosition annotation is open, so I have provided a link in the comment.